### PR TITLE
Update Thales TVL calculation across chains

### PR DESCRIPTION
**Pull request by LeifuChen**
Update TVL calculations for the Thales protocol (including Overtime V2) across multiple chains (Optimism, Arbitrum, Polygon, and Base)

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/thales exports TVL: 
        
 
 
    staking                   9.25 M
    optimism-staking          6.72 M
    optimism                  2.86 M
    pool2                     2.64 M
    arbitrum                  1.88 M
    arbitrum-staking          1.35 M
    base-staking              1.19 M
    optimism-pool2            1.16 M
    base-pool2                1.12 M
    arbitrum-pool2            363.89 k
    base                      51.63 k
    polygon                   9.30 k
    
    total                    4.81 M

File: projects/thales/utils.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        utils.getMarketsFromManager(api, manager, 'getActiveTickets')
+      ))
+    ])
+
+    return [...digitalMarkets.flat(), ...sportsMarkets.flat()]
+  },
+
+  // Calculate pool2 TVL
+  getPool2Tvl: async (api, chain) => {
+    const config = CHAIN_CONFIG[chain].pool2
```
Comment (g1nt0ki) [inline projects/thales/utils.js:46]: thanks for the PR but pool2 is, value of LP tokens deposited in your contract (usually masterchefs) not value of tokens on LP, it belongs to dex

File: projects/thales/utils.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+const utils = {
+  // Get token address from chain config
+  getTokenAddress: (chain, symbol) => CHAIN_CONFIG[chain].tokens[symbol],
+
+  // Add LP TVL
+  addLPTvl: async (api, contract, token) => {
+    const tokenAddress = utils.getTokenAddress(api.chain, token)
+    api.add(tokenAddress, await api.call({
+      target: contract,
+      abi: abi.totalDeposited
```
Comment (g1nt0ki) [inline projects/thales/utils.js:14]: where does the deposited tokens go? I dont see it on the contract

File: projects/thales/utils.js (hunk 1 above)
Comment (LeifuChen) [reply projects/thales/utils.js:46]: Thank you for the clarification. I removed the pool2 related logic.

**Comment by llamatester** [comment]
The adapter at projects/thales exports TVL: 
        
 
 
    staking                   9.21 M
    optimism-staking          6.68 M
    optimism                  2.96 M
    arbitrum                  2.10 M
    arbitrum-staking          1.34 M
    base-staking              1.19 M
    base                      51.49 k
    polygon                   9.32 k
    
    total                    5.12 M

File: projects/thales/utils.js (hunk 2 above)
Comment (LeifuChen) [reply projects/thales/utils.js:14]: Good question! The AMM LP tokens are deposited into round pool contracts, which are independently created for each new round. 
To verify the actual token balance against `totalDeposited`:

1. Get the current round via `round`
2. Check these contract addresses' balances:
   - Current round pool: roundPools[round]
   - Next round pool: roundPools[round + 1] (for pre-deposits)
   - Default round pool: roundPools[1] (for cross-round trades)

**Pull request merged**